src/test/java/org/opensearch/geospatial/ip2geo/action/DeleteDatasourceRequestTests.java

```diff
@@ -27,4 +29,16 @@ public void testStreamInOut_whenValidInput_thenSucceed() {
         // Verify
         assertEquals(request.getName(), copiedRequest.getName());
     }
+
+    public void testValidate_whenNullOrBlank_thenError() {
+        String datasourceName = Randomness.get().nextInt() % 2 == 0 ? null : " ";
```

Comment:
I've seen in core such randomized test cases they do in multiple runs, so chance is high that during one run variable will take all possible values. Like a simple for loop with 5 to 10 iterations.